apps/hash-graph/lib/graph/src/store/postgres/ontology/entity_type.rs

```diff
@@ -288,14 +288,17 @@ impl<C: AsClient> PostgresStore<C> {
     fn create_closed_entity_type(
         entity_type_id: EntityTypeId,
         available_types: &mut HashMap<EntityTypeId, raw::EntityType>,
-    ) -> Result<raw::EntityType, QueryError> {
+        mut inheritance_depths: HashMap<EntityTypeId, i64>,
+    ) -> Result<(raw::EntityType, HashMap<EntityTypeId, i64>), QueryError> {
         let mut current_type = available_types
             .remove(&entity_type_id)
             .ok_or_else(|| Report::new(QueryError))
             .attach_printable("entity type not available")?;
-        let mut visited_ids = HashSet::from([entity_type_id]);
+        // let mut visited_ids = HashMap::from([(entity_type_id, 0)]);
+        let mut current_depth = 0;
 
```

Comment:
Even if this _should_ work, for readabilities sake I'd like to see a "real" queue instead, like:

```rust
struct Item {
	depth: u8,
	id: EntityTypeId
}

let mut queue = Vec::new();

while let Some(item) = queue.pop() {
  /* parent logic */
}
```

while the current solution (ab)uses the merging of parents, this should make it clearer as to what's happening.

Reply:
I'm not sure if this would make it necessarily easier to read. The queue would need to be populated after each iteration, so you would not avoid the outer `loop` statement. The queue would just be moved from one location to another location. Currently, it's `for parent in current_type.all_of.elements.clone()` and this would be replaced with a `while` loop.